src/overlays/actors/ovl_Dm_Hina/z_dm_hina.c

```diff
@@ -33,24 +33,156 @@ ActorInit Dm_Hina_InitVars = {
     (ActorFunc)DmHina_Draw,
 };
 
-#endif
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Hina/DmHina_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Hina/DmHina_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Hina/func_80A1F470.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Hina/func_80A1F56C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Hina/func_80A1F5AC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Hina/func_80A1F63C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Hina/func_80A1F75C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Hina/DmHina_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Hina/func_80A1F9AC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Hina/DmHina_Draw.s")
+void DmHina_Init(Actor* thisx, PlayState* play) {
+    DmHina* this = THIS;
+
+    this->unk180 = 1;
```

Comment:
```suggestion
    this->isDrawn = true;
```
Going along with engineer's recommendation, this is a bool and can be set to true.